frontend/src/pages/UsersPageServices/UsersPageService.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { useEffect } from 'react'
+import { useParams } from 'react-router-dom'
+import { findService } from '../../models/devices'
+import { useSelector } from 'react-redux'
+import { ApplicationState } from '../../store'
+import { Container } from '../../components/Container'
+import { useHistory } from 'react-router-dom'
+import analytics from '../../helpers/Analytics'
+import { SharedUsersList } from '../../components/SharedUsersList'
+import { SharedUsersHeader } from '../../components/SharedUsersHeader'
+
+export const UsersPageService: React.FC = () => {
+  const { serviceID = '' } = useParams()
+  const { deviceID = '' } = useParams()
+  const { device, service } = useSelector((state: ApplicationState) => ({
+    device: state.devices.find((d: IDevice) => d.id === deviceID),
+    service: findService(state.devices.all, serviceID)[0]
```

Comment:
the `findService` method will return an array of [service, device] so you don't need to do the device find as well on the line above. just do the select and return... like this:
`  const [service, device] = useSelector((state: ApplicationState) => findService(state.devices.all, serviceID))`
You also don't need to get the deviceID from useParams() - this will better support the connection page link change in the Router.

Reply:
good catch, thanks